automation/provisioning/ssh-keyfiles.sh

```diff
@@ -0,0 +1,99 @@
+#
+# Script
+#   ssh-keyfiles.sh
+#
+# Description
+#   Overwrites default keyfiles for ssh and restarts the ssh service.
+#
+#   Note that FTPS (FTP over SSL) is not supported.
+#
+# Usage
+#   ssh-keyfiles.sh type privatekeyfile publickeyfile
+#       type:           Required. Specify DSA, ECDSA, ED25519, or RSA. Will update the corresponding keyfile pair.
+#       privatekeyfile: Required. A base64 encoded string that provides the private keyfile data.
+#       publickeyfile:  Required. A base64 encoded string that provides the public keyfile data.
+#
+
+scriptName='ssh-keyfiles.sh'
+echo "[$scriptName] --- start ---"
+
+type=$1
+privatekeyfile=$2
+publickeyfile=$3
+
+if [ ${#privatekeyfile} -le 30 ]
+then
+    privatekeyfileTrunc="$privatekeyfile"
+else
+    privatekeyfileTrunc="$(echo "$privatekeyfile" | cut -c 1-27)...";
+fi
+
+if [ ${#publickeyfile} -le 30 ]
+then
+    publickeyfileTrunc="$publickeyfile"
+else
+    publickeyfileTrunc="$(echo "$publickeyfile" | cut -c 1-27)...";
+fi
+
+if [ "$type" = "DSA" ]
+then
+    privatekeyfileName='ssh_host_dsa_key'
+    publickeyfileName='ssh_host_dsa_key.pub'
+elif [ "$type" = "ECDSA" ]
+then
+    privatekeyfileName='ssh_host_ecdsa_key'
+    publickeyfileName='ssh_host_ecdsa_key.pub'
+elif [ "$type" = "ED25519" ]
+then
+    privatekeyfileName='ssh_host_ed25519_key'
+    publickeyfileName='ssh_host_ed25519_key.pub'
+elif [ "$type" = "RSA" ]
+then
+    privatekeyfileName='ssh_host_rsa_key'
+    publickeyfileName='ssh_host_rsa_key.pub'
+fi
+
+echo "[$scriptName] SSH Keyfile Parameters"
+echo "  type:               $type"
+echo "  privatekeyfile:     $privatekeyfileTrunc"
+echo "  publickeyfile:      $publickeyfileTrunc"
+echo "  privatekeyfileName: $privatekeyfileName"
+echo "  publickeyfileName:  $publickeyfileName"
+    
+if [ -z "$type" ] || [ -z "$privatekeyfile" ] || [ -z "$publickeyfile" ]
+then
+    >&2 echo "[$scriptName] The arguments for type, privatekeyfile and publickeyfile are all required. Please refer to $scriptName inline documentation for usage"
+    exit 1
+fi
+
+if [ "$type" != "DSA" ] && [ "$type" != "ECDSA" ] && [ "$type" != "ED25519" ] && [ "$type" != "RSA" ]
+then
+    >&2 echo "[$scriptName] The supplied value of \"$type\" for type is invalid. Only the following values are accepted: DSA, ECDSA, ED25519, RSA"
+    exit 1
+fi
+
+privatekeyfilePath="/etc/ssh/$privatekeyfileName"
```

Comment:
set /etc/ssh as a default value for an additional argument. So that if someone is creating their own certificate, i.e. don't have root access, they can set a personal path.